scripts/in_progress/regrid.py

```diff
@@ -0,0 +1,405 @@
+        # writing weight matrix to disk
+        # for storing weight matrix to other location
+        if 'dir_wm_new' in kwargs.keys():
+            # check if folder exists
+            if os.path.exists(kwargs['dir_wm_new']) is False:
+                os.makedirs(kwargs['dir_wm_new'])
+            else:
+                pass
+            wm = os.path.join(kwargs['dir_wm_new'], wm)
+        wm_ds.to_netcdf(path = wm)
+        print('done')
+    # Perform regridding 
+    print('regridding...')
+    ds_in_re = regridder(ds_in, keep_attrs = True)
+    # Update grid attrs in regridded dataset
+    ds_in_re.attrs['output_grid'] = ds_out.attrs['input_grid']
+    print('done')
+    return ds_in_re
+
+# Interpolate to WRF24 grids
+def to_WRF_grid(ds, wrf_proj):
+    """
+    Function: Interpolate input dataset (WRF/ERA5/CERES/Rutgers) to WRF24 grid and perform remapping 
+              remap from polar to WRF24 for i.ie. Rutgers Northern Hemisphere 24 km Weekly Snow Cover Extent
+              or from WGS1984 to WRF24 for i.e. ERA5 Land, ERA5 plev, ERA5 slev
+    Input: WRF / ERA5 / CERES / Rutgers xarray dataset/dataarray,
+           WRF projection, coordinate system 
+    Output: xarray dataset/dataarray in WRF24 grid
+    """
+    # Transform to WRF Projection
+    xform_pts = wrf_proj['cart_proj'].transform_points(wrf_proj['wrf_crs'],to_np(ds.lon.values),to_np(ds.lat.values))
+    wrf_x = xform_pts[...,0]
+    wrf_y = xform_pts[...,1]
+    ds_wrf=ds.assign_coords({'lat':(('south_north','west_east'),wrf_y),
+                                                       'lon':(('south_north','west_east',),wrf_x)})
+    # update attrs
+    ds_wrf.attrs['output_grid'] = 'wrf24'
+    return ds_wrf
+
+def find_tem_freq(ds_in):
+    """
+    Functions: identify temporal frequency
+    Input: 
+    1. ds_in: xarray dataset/dataarray
+    Output:
+    !. temporal frequency as string
+    """
+    freq = xr.infer_freq(ds_in.time)
+    # if xr.infer_freq does not work
+    if freq is None:
+        delta = pd.to_timedelta(np.diff(ds_in.time)).mean()
+        if delta.days <=31 and delta.days>=28:
+            freq = 'MS'
+        elif delta.days == 7:
+            freq = 'W'
+    else:
+        pass
+    # return freq name based on results
+    if 'MS' in freq or 'ME' in freq:
+        ds_freq = 'monthly'
+    elif 'W' in freq:
+        ds_freq = 'weekly'
+    elif 'Q' in freq:
+        ds_freq = 'seasonal'
+    elif 'Y' in freq:
+        ds_freq = 'yearly'
+    return ds_freq
+
+# Function to build filename of exported netcdf
+def build_regridded_nc_filename(ds_in):
+    """
+    Function: build filename for exported regridded nc
+    Inputs:
+    1. ds_in: xarray dataset/dataarray of output dataset with ds_name, input_grid, output_grid, ds_path as attrs
+    Output:
+    Filename
+    """
+    # for WRF
+    if re.search('wrf', ds_in.attrs['ds_path'], re.I):
+        # for WRF norm
+        if re.search('-norm-', ds_in.attrs['ds_path']):
```

Comment:
if you end up reusing an attribute all the time, it is often convenient to store it in a variable (it is probably also more efficient, but that is not a major concern here).